generic-server/src/main.rs

```diff
@@ -36,3 +48,25 @@ fn config_tracing() {
         .with(filter)
         .init();
 }
+
+#[cfg(test)]
+mod tests {
+    use tokio::signal;
+
+    use super::run_and_shutdown_server;
+    use std::net::SocketAddr;
+
+
+    #[tokio::test]
+    async fn test_server_shutdown() {
+        let port = std::env::var("SERVER_LOCAL_PORT").unwrap_or("3000".to_owned());
+        let addr: SocketAddr = format!("127.0.0.1:{port}").parse().unwrap();
+        tracing::info!("listening on {addr}");
+
+        // run server in background
+        tokio::spawn(run_and_shutdown_server(addr));
+        // send a shutdown signal to main thread
+        signal::unix::SignalKind::terminate();
+
+            }
```

Comment:
Do something: Comment out the line that sends the terminate signal. Does your test fail? On my side, it is NO. We would then agree that it is not doing what it pretends to. 

In fact, I think it will likely not be easy to write. You can skip it if it eats up too much of your time.